Update getting-started example to use new drive paradigm

WardBenjamin opened the pull request:
This removes the deprecated LiveWindow.run() usage, as per [this allwpilib example](https://github.com/wpilibsuite/allwpilib/blob/0ef9803363380c58804a0d885a7cc7ea8233a1ac/wpilibcExamples/src/main/cpp/examples/GearsBot/src/Robot.cpp), the new `arcadeDrive` syntax as per [this allwpilib example](https://github.com/wpilibsuite/allwpilib/blob/master/wpilibcExamples/src/main/cpp/examples/ArcadeDrive/src/Robot.cpp), and uses the new `DifferentialDrive` class instead of the deprecated `RobotDrive`.

Note that this removes parity with the ScreenSteps documentation, since it has not yet been updated for both C++ and Java. I've already sent an email about this.

I'm also just about finished with updating the drivetrain sections of the Anatomy of a Robot page from the Programmer's Guide.

WardBenjamin commented:
Also note that the Travis CI build is using (no-longer-supported-by-pyfrc) Python 3.4.

auscompgeek commented:
Yeah, I have an open PR about dropping Python 3.4 from the tests here…

auscompgeek commented on getting-started/robot.py:
```diff
@@ -21,21 +25,22 @@ def autonomousInit(self):
         """This function is called periodically during test mode."""
-        wpilib.LiveWindow.run()
+        pass
+
```
Hm. We don't have a `robotPeriodic`, `teleopInit`, nor `testInit`, so why not just delete this unnecessary method?

WardBenjamin replied:
I left this in to mirror the allwpilib example linked in the first post.

auscompgeek replied:
You linked the C++ gearsbot example. We're talking about the getting started example…

If you really wanted to mirror the getting started ([C++](https://github.com/wpilibsuite/allwpilib/blob/v2018.1.1/wpilibcExamples/src/main/cpp/examples/GettingStarted/src/Robot.cpp) and [Java](https://github.com/wpilibsuite/allwpilib/blob/v2018.1.1/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/gettingstarted/Robot.java), but you really should be looking at the Java one as we mirror WPILibJ, not WPILibC) example, you'd add in more stubs as well.

auscompgeek replied:
Speaking of which, we should probably change the autonomous counter thing to use Timer instead, like in https://github.com/wpilibsuite/EclipsePlugins/commit/f4131c91ac01cd21bac36c8372d0ab7046d3c4b2#diff-c0b3b6a7b702a9920b95516dac2c0c7c.

WardBenjamin commented:
@auscompgeek I agree, I'll get on that now. I'll also open another PR to the docs repo to keep them even.

virtuald commented:
If you rebase against master I think your tests will pass.

WardBenjamin commented:
Rebased and updated to use `Timer` as per the example integrated into the Eclipse plugin. The ScreenSteps examples still aren't updated, last time I checked.

virtuald commented:
Thanks!
